src/directory/footer.rs

```diff
@@ -1,69 +1,45 @@
-use crate::common::{BinarySerializable, CountingWriter, FixedSize, HasLen, VInt};
 use crate::directory::error::Incompatibility;
 use crate::directory::FileSlice;
-use crate::directory::{AntiCallToken, TerminatingWrite};
-use crate::Version;
+use crate::{
+    common::{BinarySerializable, CountingWriter, DeserializeFrom, FixedSize, HasLen},
+    directory::{AntiCallToken, TerminatingWrite},
+    Version, INDEX_FORMAT_VERSION,
+};
 use crc32fast::Hasher;
+use serde::{Deserialize, Serialize};
 use std::io;
 use std::io::Write;
 
-const FOOTER_MAX_LEN: usize = 10_000;
+const FOOTER_MAX_LEN: u32 = 50_000;
+
+/// The magic byte of the footer to identify corruption
+/// or an old version of the footer.
+const FOOTER_MAGIC_BYTE: u32 = 1337;
```

Comment:
that's not single byte. FOOTER_MAGIC_NUMBER?

Reply:
yes that's better